ansible/playbooks/pre-cluster.yaml

```diff
@@ -72,7 +79,17 @@
       loop_var: idx
     when: inventory_hostname in groups.hana
 
-
+  - name: Ensure hostnames are preserved [aws]
+    ansible.builtin.lineinfile:
+      path: /etc/cloud/cloud.cfg
+      regexp: '^preserve_hostname:'
+      line: 'preserve_hostname: true'
+      mode: '0544'
+      owner: root
+      group: root
+      create: yes
+      backup: yes
+    when: cloud_platform_is_aws
```

Comment:
Where `cloud_platform_is_aws` come from?

Reply:
Yes.  detect os is run as a pre-task and that exposes its variables to this playbook.